api/lib/application/users/index.js

```diff
@@ -325,9 +305,16 @@ const register = async function (server) {
             userId: identifiersType.userId,
             authenticationMethodId: identifiersType.authenticationMethodId,
           }),
-          payload: reassignAuthenticationMethodJoiSchema,
+          payload: Joi.object({
+            data: {
+              attributes: {
+                'user-id': identifiersType.userId,
+              },
```

Comment:
C'est bien de retirer le schema Joi avec le lien contenant les constantes, mais je pense qu'il serait mieux de retirer le `allowUnknown: true` et de remettre `identity-provider: Joi.string().required()`.

```suggestion
              attributes: {
                'user-id': identifiersType.userId,
                'identity-provider': Joi.string().required()
              },
```

Reply:
Une raison, je pense, qui a fait que le `allowUnknown: true` est que le champ identity-provider n'est pas du tout utilisé côté [controller](https://github.com/1024pix/pix/blob/f41dc5b1d2249662fc7e199356db2a9c8c097ae1/api/lib/application/users/user-controller.js#L368-L379).
C'est le paramètre `authenticationMethodId` qui est utilisé.

Ce qui fait qu'on pourrait potentiellement l'enlever, mais cela implique des modification côté PixAdmin pour ne plus passer le paramètre 🤔 

D'où la présence de ce champ en attendant 😄